code/addons/essentials/src/docs/preset.ts

```diff
@@ -5,7 +5,7 @@ export * from '@storybook/addon-docs/dist/preset';
 export const mdxLoaderOptions = async (config: any) => {
   config.mdxCompileOptions.providerImportSource = join(
     dirname(require.resolve('@storybook/addon-docs/package.json')),
-    '/dist/shims/mdx-react-shim'
+    '/dist/shims/mdx-react-shim.mjs'
```

Comment:
This exact line is causing a bug for me.
I was wondering why do you even need to override `providerImportSource` in this `@storybook/addon-essentials` preset? I mean wouldn't `providerImportSource` be covered for webpack and vite builder by 2 other files? Maybe instead of fixing it by providing the extension it's better to remove this entire override in this file?

Reply:
@ndelangen could you provide the context here? Thanks!

Reply:
Uhm... don't have the context why it's needed, though I think @JReinhold probably does.

Reply:
I agree that this doesn't look necessary to me. You can try removing it and see if CI still works.